decentralized-message-queue/test/Test/DMQ/Protocol/LocalMsgNotification.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE FlexibleContexts, FlexibleInstances #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.DMQ.Protocol.LocalMsgNotification where
+
+import Codec.Serialise (DeserialiseFailure)
+import Control.Monad.Class.MonadST (MonadST)
+import Control.Monad.ST (runST)
+import Data.ByteString.Lazy (ByteString)
+import Data.List.NonEmpty qualified as NonEmpty
+import Test.QuickCheck qualified as QC
+import Test.Tasty
+import Test.Tasty.QuickCheck
+
+import DMQ.Protocol.LocalMsgNotification.Codec
+import DMQ.Protocol.LocalMsgNotification.Type
+import DMQ.Protocol.SigSubmission.Codec
+import DMQ.Protocol.SigSubmission.Type hiding (SingBlockingStyle (..), BlockingReplyList (..))
+import Network.TypedProtocol.Codec hiding (prop_codec)
+import Test.DMQ.Protocol.SigSubmission ()
+import Test.Ouroboros.Network.Protocol.Utils
+import Test.Ouroboros.Network.Utils
+
+
+tests :: TestTree
+tests =
+  testGroup "DMQ.Protocol"
+    [ testGroup "LocalMsgNotification"
+      [ testProperty "codec" prop_codec
+      , testProperty "codec 2-splits" prop_codec_splits2
+      , testProperty "codec 3-splits" $ withMaxSize 10
+                                          prop_codec_splits3
+      , testProperty "codec cbor"       prop_codec_cbor
+      , testProperty "codec valid cbor" prop_codec_valid_cbor
```

Comment:
We're also missing some other tests:

* `prop_direct_{ST,IO}`
* `prop_connect_{ST,IO}`
* `prop_channel_{ST,IO}`